api/schemas/project.py

```diff
@@ -18,4 +18,41 @@ class Project(BaseModel):
     customer_id: int
     customer_name: Optional[str] = None
     area_id: int
+
+    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)
+
+
+class ProjectAllocation(BaseModel):
+    id: int
+    user_id: int
+    project_id: int
+    username: Optional[str] = None
+    start_date: Optional[date] = None
+    end_date: Optional[date] = None
+    hours_per_day: Optional[float] = None
+    total_hours: Optional[float] = None
+    fte: Optional[float] = None
+    is_tentative: Optional[bool] = None
+    is_billable: Optional[bool] = None
+    notes: Optional[str] = None
+
     model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)
+
+
+class NewProjectAllocation(BaseModel):
```

Comment:
If all the properties are the same as the ProjectAllocation model, I think you can inherit. So `NewProjectAllocaiton(ProjectAllocation)` and then `pass` in the body. You would then only need to specify a property in the New model if it differed from the ProjectAllocation (i.e., if it was now required instead of optional).

Reply:
Good catch, they hadn't exactly the same properties but could be cleaned up indeed.